service/worker/deployment/deployment_workflow.go

```diff
@@ -191,7 +151,70 @@ func (d *DeploymentWorkflowRunner) run() error {
 
 	d.logger.Debug("Deployment doing continue-as-new")
 	return workflow.NewContinueAsNewError(d.ctx, DeploymentWorkflow, d.DeploymentWorkflowArgs)
+}
+
+func (d *DeploymentWorkflowRunner) validateRegisterWorker(args *deploymentspb.RegisterWorkerInDeploymentArgs) error {
```

Comment:
nice!

Reply:
just pasting this here since this is something I learned and might help somebody - I was curious on understanding what would happen if the validator were to be called before the function conducting the update was - to the naked eye, this would panic since there could be a chance `args.TaskQueue` may not have a value present in the map

turns out all panics from validator function will be returned as errors which is sweet since nothing shall break

Reply:
The validator is called before the update, that's the whole point.

If args.TaskQueueName isn't in the map, the lookup will return nil. GetTaskQueues() on nil returns nil, and a lookup on that nil map will return not found. It will not panic.

Reply:
> lookup on that nil map will return not found.

I thought over here, we would receive a panic since we could be trying to do a `nil` lookup on a `nil` map right - my apologies if my assumption was wrong...